clio-integration-test/src/it/scala/org/broadinstitute/clio/integrationtest/tests/GvcfTests.scala

```diff
@@ -375,4 +375,172 @@ trait GvcfTests { self: BaseIntegrationSpec =>
       }
     }
   }
+
+  it should "move gvcfs in GCP" in {
+    val project = s"abcdefg$randomId"
+    val sample = s"sample$randomId"
+    val version = 3
+
+    val fileContents = s"$randomId --- I am a dummy gvcf --- $randomId"
+    val cloudPath = rootTestStorageDir.resolve(
+      s"gvcf/$project/$sample/v$version/$randomId.vcf.gz"
+    )
+    val cloudPath2 = cloudPath.getParent.resolve(s"moved/$randomId.vcf/gz")
+
+    val key = TransferGvcfV1Key(Location.GCP, project, sample, version)
+    val metadata =
+      TransferGvcfV1Metadata(gvcfPath = Some(cloudPath.toUri.toString))
+
+    // Clio needs the metadata to be added before it can be moved.
+    val _ = Files.write(cloudPath, fileContents.getBytes)
+    val result = for {
+      _ <- runUpsertGvcf(key, metadata)
+      _ <- runClient(
+        ClioCommand.moveGvcfName,
+        "--location",
+        Location.GCP.entryName,
+        "--project",
+        project,
+        "--sample-alias",
+        sample,
+        "--version",
+        version.toString,
+        "--destination",
+        cloudPath2.toUri.toString
+      )
+    } yield {
+      Files.exists(cloudPath) should be(false)
+      Files.exists(cloudPath2) should be(true)
+      new String(Files.readAllBytes(cloudPath2)) should be(fileContents)
+    }
+
+    result.andThen[Unit] {
+      case _ => {
+        val _ = Seq(cloudPath, cloudPath2).map(Files.deleteIfExists)
+      }
+    }
+  }
+
```

Comment:
Why not just ...
``` scala
result.andThen {
  case _ =>
    Seq(cloudPath, cloudPath2).map(Files.deleteIfExists)
}
```
... and so on.
Or are you just trying to be REALLY explicit about discarding the Booleans?

Reply:
Being really explicit because the compiler will error with non-Unit discarded value.